### DISCUSSION TITLE
Ampache5 installation: immediate php TypeError

### DISCUSSION THREAD

**Discussion by sgtell** [ampache/ampache#3415] [Q&A] [answered]
I've used ampache version 3.x for years, now I've upgraded my home server box (Alma Linux 9.1), and am trying to get to 5.5.6 or somthing current.
After getting VirtualHost configured in apache, I browse to somthing like this to get started:
http://ampache/index.php
which gets redirected to http://ampache.telltronics.org/test.php   (so I know the index.php is getting started) but then 
I got a blank screen, and error 500 (internal server error) in the logs.
Enabling "display_errors = On" gets me to this error message:

```
Warning: preg_match(): Compilation failed: unrecognised compile-time option bit(s) at offset 0 in /home/httpd/ampache/ampache5/vendor/nyholm/psr7-server/src/ServerRequestCreator.php on line 284

Warning: preg_replace_callback(): Compilation failed: unrecognised compile-time option bit(s) at offset 0 in /home/httpd/ampache/ampache5/vendor/nyholm/psr7/src/Uri.php on line 296

Fatal error: Uncaught TypeError: Nyholm\Psr7\Uri::filterPath(): Return value must be of type string, null returned in /home/httpd/ampache/ampache5/vendor/nyholm/psr7/src/Uri.php:296 Stack trace: #0 /home/httpd/ampache/ampache5/vendor/nyholm/psr7/src/Uri.php(192): Nyholm\Psr7\Uri->filterPath() #1 /home/httpd/ampache/ampache5/vendor/nyholm/psr7-server/src/ServerRequestCreator.php(294): Nyholm\Psr7\Uri->withPath() #2 /home/httpd/ampache/ampache5/vendor/nyholm/psr7-server/src/ServerRequestCreator.php(163): Nyholm\Psr7Server\ServerRequestCreator->createUriFromArray() #3 /home/httpd/ampache/ampache5/vendor/nyholm/psr7-server/src/ServerRequestCreator.php(80): Nyholm\Psr7Server\ServerRequestCreator->getUriFromEnvWithHTTP() #4 /home/httpd/ampache/ampache5/vendor/nyholm/psr7-server/src/ServerRequestCreator.php(71): Nyholm\Psr7Server\ServerRequestCreator->fromArrays() #5 /home/httpd/ampache/ampache5/public/install.php(33): Nyholm\Psr7Server\ServerRequestCreator->fromGlobals() #6 {main} thrown in /home/httpd/ampache/ampache5/vendor/nyholm/psr7/src/Uri.php on line 296
```

This is on Alma 9.1 (RHEL 9.1 clone), with php-8.0.20-3.el9.x86_64
(I can paste in a php info() dump if it would help)

help!  I can't even go back to 3.7; it also throws errors on this php.

**Comment by lachlan-00** [comment]
is it a release zip? depending on the php version too it might need a composer update.

Blank screen is usually missing or bad composer dependancies

**Comment by sgtell** [comment]
The install is ampache-5.5.6_all_php8.0.zip; my understanding is that those are supposed to include all of their dependencies?
composer (composer-2.4.4-1.el9.noarch)  fails with:
```

$ composer install
PHP Warning:  PHP Startup: ^(text/|application/xhtml\+xml) (offset=0): unrecognised compile-time option bit(s) in Unknown on line 0
PHP Fatal error:  Uncaught ErrorException: preg_match_all(): Compilation failed: unrecognised compile-time option bit(s) at offset 0 in /usr/share/php/Composer/vendor/symfony/console/Formatter/OutputFormatter.php:152
Stack trace:
#0 [internal function]: Composer\Util\ErrorHandler::handle()
#1 /usr/share/php/Composer/vendor/symfony/console/Formatter/OutputFormatter.php(152): preg_match_all()
#2 /usr/share/php/Composer/vendor/symfony/console/Formatter/OutputFormatter.php(135): Symfony\Component\Console\Formatter\OutputFormatter->formatAndWrap()
#3 /usr/share/php/Composer/vendor/symfony/console/Output/Output.php(157): Symfony\Component\Console\Formatter\OutputFormatter->format()
#4 /usr/share/php/Composer/vendor/symfony/console/Output/Output.php(132): Symfony\Component\Console\Output\Output->write()
#5 /usr/share/php/Composer/vendor/symfony/console/Application.php(836): Symfony\Component\Console\Output\Output->writeln()
#6 /usr/share/php/Composer/vendor/symfony/console/Application.php(154): Symfony\Component\Console\Application->renderThrowable()
#7 /usr/share/php/Composer/vendor/symfony/console/Application.php(177): Symfony\Component\Console\Application->Symfony\Component\Console\{closure}()
#8 /usr/share/php/Composer/Console/Application.php(139): Symfony\Component\Console\Application->run()
#9 /usr/bin/composer(89): Composer\Console\Application->run()
#10 {main}
  thrown in /usr/share/php/Composer/vendor/symfony/console/Formatter/OutputFormatter.php on line 152


```

**Comment by sgtell** [comment, marked as answer]
A web search suggests that ` preg_match_all(): Compilation failed: unrecognised compile-time option bit(s)`  is related to missing libpcre.  while libpcre2 is installed, upgrading it seems to help a lot.
Upgraded:
  pcre2-10.40-2.el9.x86_64     pcre2-devel-10.40-2.el9.x86_64     pcre2-syntax-10.40-2.el9.noarch     pcre2-utf16-10.40-2.el9.x86_64     pcre2-utf32-10.40-2.el9.x86_64  

Then composer looks better:
  $ composer install
Loading "robloach/component-installer" which is a legacy composer-installer built for Composer 1.x, it is likely to cause issues as you are running Composer 2.x.
Installing dependencies from lock file (including require-dev)
Verifying lock file contents can be installed on current platform.
Nothing to install, update or remove
Package nabil1337/case-helper is abandoned, you should avoid using it. Use marcusball/case-helper instead.
Package robloach/component-installer is abandoned, you should avoid using it. Use oomphinc/composer-installers-extender instead.
Package php-cs-fixer/diff is abandoned, you should avoid using it. No replacement was suggested.
Generating autoload files
> @php resources/scripts/composer/check_tag_tools.php

and I get  "choose installation language" instead of error messages from ampache/install.php !

(Thanks for the suggestions; hope this helps somebody else!)
